src/Common/AsynchronousMetrics.cpp

```diff
@@ -78,6 +79,7 @@ AsynchronousMetrics::AsynchronousMetrics(
         openFileIfExists("/sys/fs/cgroup/memory.current", cgroupmem_usage_in_bytes);
     }
     openFileIfExists("/sys/fs/cgroup/cpu.max", cgroupcpu_max);
+    openFileIfExists("/sys/fs/cgroup/cpu.stat", cgroupcpu_stat);
```

Comment:
This refers to the top-level cgroup. If ClickHouse runs (for whatever reason) in a nested cgroup, then `/sys/fs/cgroup/cpu.stat` might include CPU usage of other cgroups as well, i.e. become inaccurate. For some reason, l. 76-81 make the same wrong assumption.

To get the right cgroup, you could use `cgroupV2PathOfProcess` (base/base/cgroupsv2.h).